qa/rpc-tests/wallet_nullifiers.py

```diff
@@ -84,15 +84,10 @@ def run_test (self):
         assert_equal(self.nodes[3].z_getbalance(myzaddr3), zsendmany2notevalue)
         assert_equal(self.nodes[2].z_getbalance(myzaddr), zaddrremaining)
 
-        # Parallel encrypted wallet can't cache nullifiers for received notes,
-        # and therefore can't detect spends. So it sees a balance corresponding
-        # to the sum of both notes it received (one as change).
-        # TODO: Devise a way to avoid this issue (#1528)
-        assert_equal(self.nodes[1].z_getbalance(myzaddr), zsendmanynotevalue + zaddrremaining)
+        # Parallel encrypted wallet can cache nullifiers for Sapling received notes
+        assert_equal(self.nodes[1].z_getbalance(myzaddr), zaddrremaining)
```

Comment:
Since the behaviour here is observably different between Sprout and Sapling, I think we should be testing both.

Reply:
The Sprout behaviour is caused by the Sprout nullifier note map not being updated for encrypted wallets; this is still checked in the `UpdateSproutNullifierNoteMap` gtest.